cogboard-webapp/src/components/Widget/index.js

```diff
@@ -133,6 +139,14 @@ const Widget = ({ id, index }) => {
     closeConfirmationDialog();
   };
 
+  const RenderedCardHeader = ({ isEmptyHeader, ...otherProps }) => {
+    if (isEmptyHeader) {
+      return <StyledEmptyCardHeader {...otherProps} />;
+    }
+
+    return <StyledCardHeader {...otherProps} />;
+  };
+
```

Comment:
Don't define React Components inside other component. It cause serious performance problems. Neither React Components nor render functions returning jsx markup.

Reply:
Moved to new component.